src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -651,7 +649,7 @@ async Task processPartialSymbolLocationsAsync(ImmutableArray<CompilationEvent> c
                             Task.Run(() =>
                             {
                                 var treeModel = _compilationData.GetOrCreateCachedSemanticModel(tree, _compilation, cancellationToken);
-                                return GetAnalyzerSemanticDiagnosticsAsync(treeModel, filterSpan: null, cancellationToken);
+                                return GetAnalyzerSemanticDiagnosticsCoreAsync(treeModel, filterSpan: null, analysisScope.Analyzers, cancellationToken);
```

Comment:
This change and the one immediately below is the primary performance optimization in this file. Previously, we accidentally ignored the input `analysisScope` on which we compute semantic diagnostics for partial trees, which forced all analyzers to execute instead of just the ones in analysis scope.